LLM_GATEWAY.md

```
@@ -0,0 +1,108 @@
+import { LLMProvider, LLMMessage, LLMResponse, LLMProviderConfig, LLMFactory } from '../index';
+
+export class NewProvider extends LLMProvider {
+  // Implementation here
+  
+  async chat(messages: LLMMessage[]): Promise<LLMResponse> {
+    // Call provider API and return result
+  }
+}
+
+// Register the provider with the factory
+LLMFactory.registerProvider('new-provider', NewProvider);
```

markdownlint-cli2 (0.18.1)

Files should end with a single newline character

(MD047, single-trailing-newline)